### PR TITLE
Added a suppressor when CA1812 fires on NUnit Test classes.

### LINKED ISSUE #565: [TestFixture] attribute suppresses `CA1812`
We try to keep our code clean so we've enabled optional analyzers including [CA1812: Avoid uninstantiated internal classes](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1812).

Whenever we have an internal class used as a test fixture, this rule always lights up giving us false positives.

Much like there's an analyzer to suppress nullability warnings when doing `Assert.That(x, Is.Not.Null);`, can we get an analyzer suppressor which disables this rule in the presence of `[TestFixture]`?

### PR DESCRIPTION
**Pull request by manfred-brands**
Fixes #565

### CODE REVIEW

File: documentation/NUnit3003.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+## Description
+
+Class is a NUnit TestFixture and called by reflection
+
+## Motivation
+
+The default roslyn analyzer has rule [CA1812](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1812)
+which warns about internal classes not being used.
+That analyzer doesn't know about NUnit test classes.
+This suppressor catches the error, verifies the class is an NUNit TestFixture and if so suppresses the error.
```
Comment (mikkelbu) [inline documentation/NUnit3003.md:22]:
```suggestion
This suppressor catches the error, verifies the class is an NUnit TestFixture and if so suppresses the error.
```

File: src/nunit.analyzers.tests/DiagnosticSuppressors/AvoidUninstantiatedInternalClassesSuppressorTests.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            // Find the NetAnalyzers assembly (note version should match the one referenced)
+            string netAnalyzersPath = Path.Combine(PathHelper.GetNuGetPackageDirectory(),
+                "microsoft.codeanalysis.netanalyzers/7.0.3/analyzers/dotnet/cs/Microsoft.CodeAnalysis.CSharp.NetAnalyzers.dll");
+            Assembly netAnalyzerAssembly = Assembly.LoadFrom(netAnalyzersPath);
+            Type analyzerType = netAnalyzerAssembly.GetType("Microsoft.CodeQuality.CSharp.Analyzers.Maintainability.CSharpAvoidUninstantiatedInternalClasses", true)!;
+            this.analyzer = (DiagnosticAnalyzer)Activator.CreateInstance(analyzerType!)!;
```
Comment (mikkelbu) [inline src/nunit.analyzers.tests/DiagnosticSuppressors/AvoidUninstantiatedInternalClassesSuppressorTests.cs:65]: Is the first `!` needed in `Activator.CreateInstance(analyzerType!)!;` given that we also have used `!` on the previous line (line 64).

**Review by mikkelbu — APPROVED** [review]
Looks good. I only have the smallest suggestion and question.

File: src/nunit.analyzers.tests/DiagnosticSuppressors/AvoidUninstantiatedInternalClassesSuppressorTests.cs (hunk 2 above)
Comment (manfred-brands) [reply src/nunit.analyzers.tests/DiagnosticSuppressors/AvoidUninstantiatedInternalClassesSuppressorTests.cs]: You are correct. Removed.

File: documentation/NUnit3003.md (hunk 1 above)
Comment (manfred-brands) [reply documentation/NUnit3003.md]: Thanks. Updated.

**Pull request merged**